StarVMC/StarAgmlLib/AgMath.h

```diff
@@ -67,10 +69,10 @@ template <typename T> T min(const T &a, const T &b ){ return TMath::Min(a,b); }
 Int_t mod(const Int_t &a, const Int_t &b);
 Float_t mod(const Float_t &a, const Float_t &b);
 
-// #define cos(x) TMath::Cos( (Double_t) x )
-// #define sin(x) TMath::Sin( (Double_t) x )
-// #define tan(x) TMath::Tan( (Double_t) x )
-// #define exp(x) TMath::Exp( (Double_t) x )
+#define cos(x) TMath::Cos( (Double_t) x )
+#define sin(x) TMath::Sin( (Double_t) x )
+#define tan(x) TMath::Tan( (Double_t) x )
+#define exp(x) TMath::Exp( (Double_t) x )
```

Comment:
Are these defines really necessary? Don't they collide with the ones from the standard library?

Reply:
I believe that I put those in place to minimize numerical differences between the ROOT and GEANT positions and / or dimensions of a fairly small number of volumes.

Reply:
Okay... a little archeology is important here.

From commit fd6f8ab786fa7bf435ba290e9059556f7c56bc24

    sin, cos, tan, and exp need to be computed at highest precision possible in order to minimize
    introduction of roundoff error in results.

There seems to be a history of people reflexively removing this code...
fc2109800a5a00e3c915b5b1c8952a8646d38d4c

... and I keep having to restore it...
202690286974776a0098f4f2ae2368c8e5e7c407

So, yes.  These definitions are considered to be necessary.  The goal being to minimize roundoff errors between the geant3 and root geometry models.  Removal would minimally require material budget-level consistency checks to be made to validate geometries.

It is worth adding some comments for posterity into AgMath.h methinks.